test/e2e/validation_test.go

```diff
@@ -273,3 +274,15 @@ func createWindowsMachineConfig(namespace string, isReplicasFieldRequired bool,
 	}
 	return wmc
 }
+
+// testWorkerLabel tests if the worker label has been applied properly
+func testWorkerLabel(t *testing.T, nodeCount int) {
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup()
+	nodes, err := framework.Global.KubeClient.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: wmc.WindowsOSLabel})
+	require.NoError(t, err, "error querying the nodes")
+	require.Equal(t, len(nodes.Items), nodeCount, "mismatched node count")
```

Comment:
@ravisantoshgudimetla I see why you want to add `nodes` to the global context. I will add it and hopefully we don't run into any staleness issues.